README.md

```diff
@@ -275,6 +275,126 @@ An example policy would look like;
 }
 ```
 
+## Message formats and AWS interactions
+
+JustSaying uses a number of Amazon Web Services APIs to transport messages between publishers and subscribers. Since these HTTP APIs can be used by applications which do not use JustSaying, it is entirely possible to have, for example, a Java application publishing a message which is subscribed to by an application using JustSaying. Equally, a C# application publishing messages using JustSaying, which are subscribed to by a Java application, is fully supported by AWS. In order to support this interoperability, it is important that the actual message formats are described. Since Amazon provide SDKs for many programming languages and frameworks, you are unlikely to interact with the APIs purely via HTTP, but knowing the structure of the JustSaying message JSON itself is useful for cross-language purposes.
+
+### Publishing to SNS
+
+JustSaying uses the SNS Publish API to send messages to SNS (documented [here](https://docs.aws.amazon.com/sns/latest/api/API_Publish.html)). The following parameters are set:
+
+#### TopicArn
+The ARN of the topic to which the message is being published.
+
+#### Subject
+JustSaying will set this to the type of the .NET message class - "OrderAccepted", for example.
+
+#### Message
+The actual message object, serialised to JSON. Because JustSaying messages must derive from the base Message class, the properties on that base class will be serialised into this JSON. On the base class, the `Id` property is set as a random GUID and the `TimeStamp` property is set to the current time (in UTC). The default serialisation settings will ignore null properties, so properties like `Conversation` will not appear in the JSON if they are left as the default null value. Enums are serialised as strings.
```

Comment:
_"Enumerations are..."?

Reply:
Again this feels C# specific. We might say that we recommend that enumerated values are sent as strings to allow easier comprehension of the message and the C# library defaults to this

Reply:
Well, if we don't describe what JustSaying (a C# library) does, then how do people not using JustSaying know how to work together? By saying that we serialise enumerations as strings when publishing, a Python/Java/whatever developer then knows what JSON they'll be getting when they subscribe.

Hmmmmm, instead of saying "this is what JustSaying does", is it better to invert it and say rephrase it as "this is what you should do to send a message to JustSaying / what you should expect when you receive a message from JustSaying"?

Reply:
I've added a table that lists the the properties in the base Message class, which ones JustSaying sets (`Id` and `TimeStamp`) and their intended use (or as best I can guess based on the names!) Also noted that properties which are _not_ set don't get serialised.